modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinDefaultTimeoutTest.java

```diff
@@ -0,0 +1,133 @@
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.jdbc.thin;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.apache.ignite.cache.query.annotations.QuerySqlFunction;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.configuration.SqlConfiguration;
+import org.apache.ignite.internal.IgniteEx;
+import org.apache.ignite.testframework.GridTestUtils;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.Test;
+
+/** */
+public class JdbcThinDefaultTimeoutTest extends GridCommonAbstractTest {
+    /** */
+    public static final int ROW_COUNT = 200;
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        CacheConfiguration<Object, Object> ccfg = new CacheConfiguration<>(DEFAULT_CACHE_NAME)
+            .setIndexedTypes(Integer.class, Integer.class)
+            .setSqlSchema("PUBLIC")
+            .setSqlFunctionClasses(TestSQLFunctions.class);
+
+        return super.getConfiguration(igniteInstanceName)
+            .setCacheConfiguration(ccfg)
+            .setSqlConfiguration(new SqlConfiguration().setDefaultQueryTimeout(100));
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        super.beforeTestsStarted();
+
+        IgniteEx ign = startGrid(0);
+
+        Map<Integer, Integer> vals = IntStream.range(0, ROW_COUNT)
+            .boxed()
+            .collect(Collectors.toMap(Function.identity(), Function.identity()));
+
+        // We need to fill cache with many rows because server-side timeout checks for timeout periodically after
+        // loading several rows.
+        ign.cache(DEFAULT_CACHE_NAME).putAll(vals);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTestsStopped() throws Exception {
+        stopAllGrids();
+
+        super.afterTestsStopped();
+    }
+
+    /** */
+    @Test
+    public void testDefaultTimeoutIgnored() throws Exception {
```

Comment:
If you wanted to verify _default timeout_, you have to use it in the test. So I could suggest a scenario like this:
1) provide function that would sleep for specified amount of time, but only once (on the very first invocation)
2) run query that will invoke the function with SqlConfiguration#DFLT_QRY_TIMEOUT as a param
3) verify result

this is very similar to what you provide except sleeping time is strongly determined and will be changed within changing of defaults itself.

This suggestion is applied and to the next test too.